packages/queued-request-controller/src/constants.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * This is a list of methods that require the globally selected network
+ * to match the dapp selected network before being processed. These can
+ * be for UI/UX reasons where the currently selected network is displayed
+ * in the confirmation even though it will be submitted on the correct
+ * network for the dapp. It could also be that a method expects the
+ * globally selected network to match some value in the request params itself.
+ */
+export const methodsRequiringNetworkSwitch = [
```

Comment:
Rather than describing these here as constants, perhaps we should accept them as controller or middleware options? The current set of supported methods, and whether each one has a confirmation or not, may differ between platforms. Listing this at the platform layer should let us reuse this list for other functionality as well, and it would perhaps make it less likely we'd forget to update this list after adding a new confirmation.

Reply:
This would also let us ship it as a non-breaking change, as we could use the existing list as a fallback